### PR TITLE
In-Memory Event bus Tests

### PR DESCRIPTION
**Pull request by kamilbaczek**

### CODE REVIEW

File: Src/Fitnet/Shared/Events/IIntegrationEvent.cs [hunk 1]
```diff
@@ -2,6 +2,10 @@ namespace SuperSimpleArchitecture.Fitnet.Shared.Events;
 
 using MediatR;
 
+internal interface IIntegrationEventHandler<in TEvent> : INotificationHandler<TEvent> where TEvent : IIntegrationEvent
```
Comment (kamilbaczek) [inline Src/Fitnet/Shared/Events/IIntegrationEvent.cs:5]: Extract file

File: Src/Fitnet.IntegrationTests/Shared/Events/EventBus/InMemory/InMemoryEventBusTests.cs [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    
+    [Fact]
+    public async Task Given_valid_event_Then_event_should_be_consumed()
+    {
+        // Arrange
+        var eventBus = _applicationInMemory.Services.GetRequiredService<IEventBus>();
+        var fakeEvent = FakeEvent.Create();
+        
+        // Act
+         await eventBus.PublishAsync(fakeEvent, CancellationToken.None);
```
Comment (meaboutsoftware) [inline Src/Fitnet.IntegrationTests/Shared/Events/EventBus/InMemory/InMemoryEventBusTests.cs:25]: Remove not needed space

File: Src/Fitnet.IntegrationTests/Shared/Events/EventBus/InMemory/InMemoryEventBusTests.cs [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    private readonly WebApplicationFactory<Program> _applicationInMemory;
+
+    public InMemoryEventBusTests(WebApplicationFactory<Program> applicationInMemoryFactory,
+        DatabaseContainer database) =>
+        _applicationInMemory = applicationInMemoryFactory
+            .WithContainerDatabaseConfigured(database.ConnectionString!)
+            .WithFakeConsumers();
+    
+    [Fact]
+    public async Task Given_valid_event_Then_event_should_be_consumed()
```
Comment (meaboutsoftware) [inline Src/Fitnet.IntegrationTests/Shared/Events/EventBus/InMemory/InMemoryEventBusTests.cs:18]: Maybe Given_valid_event_**published**_Then...?

Especially as we will have Subscribe as well in this class, then it makes sense to distinguish it.

Btw. That's why I prefer one of these approaches (we won't change it but as info) either:

Should...When - ShouldBeConsumed_WhenValidEventPublished
Given...When...Then - GivenValidEvent_WhenPublished_ThenItShouldBeConsumed

:)

File: Src/Fitnet/Shared/Events/IIntegrationEvent.cs (hunk 1 above)
Comment (meaboutsoftware) [reply Src/Fitnet/Shared/Events/IIntegrationEvent.cs:5]: Why do you think it makes sense? :) Both are completely related to each other

**Review by meaboutsoftware — CHANGES_REQUESTED** [review]

File: Src/Fitnet/Shared/Events/IIntegrationEvent.cs (hunk 1 above)
Comment (kamilbaczek) [reply Src/Fitnet/Shared/Events/IIntegrationEvent.cs:5]: File name is IIntegrationEvent

File: Src/Fitnet.IntegrationTests/Shared/Events/EventBus/InMemory/InMemoryEventBusTests.cs (hunk 3 above)
Comment (kamilbaczek) [reply Src/Fitnet.IntegrationTests/Shared/Events/EventBus/InMemory/InMemoryEventBusTests.cs]: @meaboutsoftware thanks for this advice. I will try this approach in another project

**Review by meaboutsoftware — APPROVED** [review]

**Pull request merged**